components/micro-gateway-core/src/main/ballerina/src/gateway/constants/constants.bal

```diff
@@ -50,6 +50,8 @@ public const string VALIDATE_KEY_SOAP_ACTION = "urn:validateKey";
 public const string KEY_VALIDATION_SERVICE_CONTEXT = "/services/APIKeyValidationService";
 public const string UTF_8 = "UTF-8";
 public const string INVALID_COOKIE = "Cookie is Invalid";
+public const string WWW_AUTHENTICATE = "WWW-Authenticate";
+public const string WWW_AUTHENTICATE_ERROR = ", error=\"invalid token\" , error_description=\"The access token expired\"";
```

Comment:
Authentication failures can happen due to several reasons. If we set the description to the header, we have to mention the correct reason for the failure. 

IMO, we do not need the `error` and `error_description` in the header since the WWW-Authenticate header is meant to contain only the challenge.

Reply:
This WWW-Authenticate header is already available in the WSO2 APIM. In the APIM, it displays both `error` and `error_description` along with the WWW-Authenticate header when authentication fails. And the `error_description` is set to "The access token expired". This implementation is exactly similar to the implementation in APIM.